installers-conda/build_installers.py

```diff
@@ -150,17 +153,73 @@
     logger.info(f"Did not read specs from {SPECS}")
 
 for spec in args.extra_specs:
-    k, *v = re.split('([<>= ]+)', spec)
+    k, *v = re.split('([<>=]+)[ ]*', spec)
     specs[k] = "".join(v).strip()
-    if k == "spyder":
-        SPYVER = v[-1]
 
+PY_VER = re.split('([<>=]+)[ ]*', specs['python'])[-1]
+SPYVER = re.split('([<>=]+)[ ]*', specs['spyder'])[-1]
+
+LOCK_FILE = DIST / f"conda-{TARGET_PLATFORM}.lock"
+TMP_LOCK_FILE = BUILD / f"conda-{TARGET_PLATFORM}.lock"
 OUTPUT_FILE = DIST / f"{APP}-{OS}-{ARCH}.{args.install_type}"
 INSTALLER_DEFAULT_PATH_STEM = f"{APP.lower()}-{SPYVER.split('.')[0]}"
 
 WELCOME_IMG_WIN = BUILD / "welcome_img_win.png"
 HEADER_IMG_WIN = BUILD / "header_img_win.png"
 WELCOME_IMG_MAC = BUILD / "welcome_img_mac.png"
+CONSTRUCTOR_FILE = BUILD / "construct.yaml"
+
+
+def _create_conda_lock():
+    definitions = {
+        "channels": [
+            CONDA_BLD_PATH,
+            "conda-forge/label/spyder_dev",
+            "conda-forge/label/spyder_kernels_rc",
+            "conda-forge"
+        ],
+        "dependencies": [k + v for k, v in specs.items()],
+        "platforms": [TARGET_PLATFORM]
+    }
+
+    logger.info("Conda lock configuration:")
+    if logger.getEffectiveLevel() <= 20:
+        yaml.dump(definitions, sys.stdout)
+
+    env_file = BUILD / "runtime_env.yml"
+    yaml.dump(definitions, env_file)
+
+    env = os.environ.copy()
+    env["CONDA_CHANNEL_PRIORITY"] = "flexible"
+
+    cmd_args = [
+        "conda-lock", "lock",
+        "--kind", "explicit",
+        "--file", str(env_file),
+        "--filename-template", str(BUILD / "conda-{platform}.lock")
+        # Note conda-lock doesn't provide output file option, only template
+    ]
+
+    run(cmd_args, check=True, env=env)
+
+    logger.info(f"Contents of {TMP_LOCK_FILE}:")
+    if logger.getEffectiveLevel() <= 20:
+        print(TMP_LOCK_FILE.read_text(), flush=True)
+
+
+def _patch_conda_lock():
+    # Replace local channel url with conda-forge and remove checksum
+    tmp_text = TMP_LOCK_FILE.read_text()
+    text = re.sub(
+        f"^{_get_conda_bld_path_url()}(.*)#.*$",
+        r"https://conda.anaconda.org/conda-forge\1",
+        tmp_text, flags=re.MULTILINE
+    )
```

Comment:
Technically it does not remove only for Spyder. Any package obtained from the local channel will begin with the conda build path url and will have to be replaced with conda-forge's url and the checksums will need to be removed. In practice, this will only apply to the external-deps (pylsp, qtconsole, spyder-kernels) and spyder for PRs; for release candidates and releases, it will only apply to spyder.